feat: os segment

nwykes opened the pull request:
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

Add osicon segment for displaying an OS specific icon

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

nwykes committed:
feat: osicon segment

nwykes committed:
feat: osicon segment

nwykes committed:
feat: osicon segment

nwykes committed:
Merge branch 'main' into feat_osicon

JanDeDobbeleer commented on segment.go:
```diff
@@ -56,6 +56,8 @@ const (
 	//Node writes which node version is currently active
 	Node SegmentType = "node"
+	//OS write os specific icon
+	OsIcon SegmentType = "osicon"
```
Can we rename this to Os? As people will be able to set any string (which could also be plain `"windows"` for example), it feels more natural that way. I would leave `WindowsIcon` etc as is as the other segments use the same context and most likely people will be adding an icon.

nwykes replied:
I originally had it as "os", but had a naming collision with the os pkg.  I'll try changing the segment name to os and call the internal struct something else.

JanDeDobbeleer replied:
That's the best solution indeed 👍🏻

JanDeDobbeleer commented on segment_osicon_test.go:
```diff
@@ -0,0 +1,16 @@
+func TestOsIcon(t *testing.T) {
+	env := new(MockedEnvironment)
+	env.On("getRuntimeGOOS", nil).Return("windows")
+	osicon := &osicon{env: env}
```
You can also assign properties to validate the exact icon, so differ between set and not set.
See [here](https://github.com/JanDeDobbeleer/oh-my-posh3/blob/main/segment_exit_test.go#L30) for example.

nwykes replied:
wil do, thanks.

JanDeDobbeleer committed:
feat: git HEAD context

JanDeDobbeleer committed:
feat: rebase support

nwykes committed:
feat: session segment updates

only display UserInfoSeparator if both username and hostname are dislayed
cleanHostName only if it ends with garbage instead of contains.
  fixes .localdomain having just .local stripped
add .localdomain to garbage

JanDeDobbeleer commented:
@nwykes first off, thanks for the contributions and coffee. You made my day yesterday yesterday and today with that 🙏🏻

A final request, can you rebase to main and remove that merge commit? I have a personal dislike against those things 😅

nwykes commented:
> @nwykes first off, thanks for the contributions and coffee. You made my day yesterday yesterday and today with that 🙏🏻
> 
> A final request, can you rebase to main and remove that merge commit? I have a personal dislike against those things 😅

least I can do, since I'm now using it in powershell in windows and zsh in Mac and linux everywhere, replacing my old powerlevel10k setup.  More coffees to come I hope.

Still working on the changes, but I'll get it rebased before submitting again.

nwykes committed:
Merge branch 'feat_osicon' of github.com:nwykes/oh-my-posh3 into feat_osicon

nwykes committed:
rename osicon to os

nwykes committed:
rename osicon to os

nwykes committed:
Merge branch 'feat_osicon' of github.com:nwykes/oh-my-posh3 into feat_osicon
